File: src/core/Result.java
package core;

import java.util.ArrayList;
import java.util.HashMap;

public class Result {

	public enum Status {
		UNDEFINED, SUCCEED, WARNING, ERROR
	}
	
	public String name;
	public ArrayList<String> values;
	public HashMap<String, String> properties;
	public Status status;	
	public ArrayList<Result> results;
	
	public Result(String name){
		
		this.name = name;
		values = new ArrayList<String>();
		properties = new HashMap<String, String>();
		status = Status.UNDEFINED;
		results = new ArrayList<Result>();
	}
	
	public int hashCode(){
		int hash = name.hashCode();
		for(String value : values){
			hash += value.hashCode();
		}
		for(String key : properties.keySet()){
			hash += key.hashCode();
			String value = properties.get(key);
			if(value != null){
				hash += value.hashCode();
			}
		}
		hash += status.toString().hashCode();
		for(Result result : results){
			hash += result.hashCode();
		}
		return hash;
	}
}

File: ruby/lib/core/result.rb

class Result

  @@Resolution = [ :UNDEFINED, :WARNING, :SUCCEED, :ERROR ]
  
  def initialize(name)
    @name = name
    @resolution = :UNDEFINED
    @messages = Array.new
    @values = {}
    @results = Array.new
  end
  
  attr_accessor :name
  attr_accessor :resolution
  attr_accessor :messages
  attr_accessor :values
  attr_accessor :results

end